subsys/pm/pm.c

```diff
@@ -69,18 +69,22 @@ static int pm_suspend_devices(void)
 		int ret;
 
 		/*
-		 * ignore busy devices, wake up source and devices with
+		 * Ignore busy devices, wake up sources, and devices with
 		 * runtime PM enabled.
 		 */
-		if (pm_device_is_busy(dev) || pm_device_state_is_locked(dev)
-		    || pm_device_wakeup_is_enabled(dev) ||
-		    ((dev->pm != NULL) && pm_device_runtime_is_enabled(dev))) {
+		if (pm_device_is_busy(dev) || pm_device_state_is_locked(dev) ||
```

Comment:
check can go here. why is `dev->pm != NULL` removed?

Reply:
The function does this check itself, originally it wasn't.

Reply:
I see. What is missing in the list is then a ready check.

Reply:
Note that such device PM model is essentially broken, so I'd strongly suggest moving out of such APIs and try using runtime PM model.

Reply:
Added the check.

I do use the PM runtime, but do not enable it on uninitialized devices (for the same reason you mentioned in other comment) which means it's not excluded from system PM.